src/Illuminate/Database/Query/Builder.php

```diff
@@ -278,6 +278,11 @@ public function addSelect($column)
 	{
 		$column = is_array($column) ? $column : func_get_args();
 
+		if (is_null($this->columns))
+		{
+			$this->columns = [ '*' ];
+		}
+
 		$this->columns = array_merge((array) $this->columns, $column);
```

Comment:
There is difference between empty columns and null columns variable - first case appears if user has passed empty array to select for some reason.
```php
$model->select([])->addSelect('test');
```
Which is a different situation from
```php
$model->addSelect('foo');
```
Shortened code will look much less readable
```php
$this->columns = array_merge(is_null($this->columns) ? ['*'] : $this->columns, $column);
```